profile/profile-layered.go

```diff
@@ -253,6 +259,22 @@ func (lp *LayeredProfile) MatchFilterLists(entity *intel.Entity) (result endpoin
 	return endpoints.NoMatch, ""
 }
 
+// MatchBypassProtection checks if the entity should be denied or permitted
+// based on some bypass protection checks.
+func (lp *LayeredProfile) MatchBypassProtection(entity *intel.Entity) (endpoints.EPResult, string) {
+	if !lp.PreventBypassing() {
+		return endpoints.NoMatch, ""
+	}
+
+	// Block firefox canary domain to disable DoH
+	if strings.ToLower(entity.Domain) == "use-application-dns.net." {
+		log.Warningf("bypass protection for firefox canary")
+		return endpoints.Denied, "Firefox canary domain"
```

Comment:
Would suggest to add more context: `blocked canary domain to prevent enabling DNS-over-HTTPS`.
`Firefox` would be in both the logs and UI anyway, plus this domain might be used by other software in the future.